### DISCUSSION TITLE
cannot find program: hostname

### DISCUSSION THREAD

**Discussion by dxi1** [matteocorti/check_ssl_cert#419] [General]
Command lines work, but not in the nagios GUI: I am getting error:  SSL_CERT UNKNOWN xxxxxxxx: cannot find program: hostname

 I have tried V2.52.0 and V2.49.0, same issue. I only copy the script check_ssl_cert onto /usr/lib64/nagios/plugins/ folder and made it executable. It is on Oracle Linux 8. bind-utils, nmap, and bc are installed. nagios was installed rpm and it is on 4.4.6. Also:

$ sudo command -v hostname
/bin/hostname

Lots of other checks are working though, including check_http, check_smtp for TLS.

Any ideas?

**Comment by matteocorti** [comment]
It's strange. I just committed a new version which prints out the ```$PATH``` used by the script in debug mode.

You can then configure the plugin with ```--debug-file /tmp/check_ssl_cert.debug``` and check the line beginning with

```
[DBG] PATH:
```

Something like

```
[DBG] PATH: /Users/corti/perl5/perlbrew/bin:/Users/corti/perl5/perlbrew/perls/perl-5.36.0/bin:/opt/local/bin:/opt/local/sbin:/usr/local/opt/ruby/bin:/usr/local/lib/ruby/gems/2.6.0/bin/:/Users/corti/.cabal/bin:/opt/local/bin:/opt/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/Users/corti/bin:/opt/homebrew/bin
```

There should also be a line printing the hostname binary:

```
[DBG] hostname: /bin/hostname
```

But I suppose that in your case it will be empty

**Comment by dxi1** [comment]
Thank you Matteo. I just found out that it was a SELinux issue. I created a
custom policy and it is working fine now.

It is a great plugin, and thank you for the super job on that.
